src/plugins/index_pattern_editor/public/components/index_pattern_editor_flyout_content.tsx

```diff
@@ -105,12 +106,22 @@ const IndexPatternEditorFlyoutContentComponent = ({
 
   const { getFields } = form;
 
-  const [{ title, allowHidden, type }] = useFormData<FormInternal>({ form });
+  // `useFormData` initially returns `undefined`,
+  // we override `undefined` with real default values from `schema`
+  // to get a stable reference to avoid hooks re-run and reduce number of excessive requests
+  const [
+    {
+      title = schema.title.defaultValue,
```

Comment:
Looks like internally `useFormData` has some before init state, where these all returns `undefined` even though default values are specified in the schema.

Initial switches between `undefined` and `""` caused some redundant effects re-run

Reply:
I didn't work with our form lib before, so not sure if there is a better approach

Reply:
I wouldn't say it's a _bug_, it is more an unpleasant side effect on how the architecture and react hook lifecycle work. I looked into it while working on https://github.com/elastic/kibana/pull/109238 but decided to keep that PR focused. It is on my radar and will see what can be done.

For you information this is how the data flow works:

1. You call the hook and ask for the value --> the field is not mounted yet so its value is `undefined`
2. In a `useEffect()` inside `<UseField />` the field mounts and say to the form: "I am here!" --> this updates the field value and the `useFormData` hook triggers an update that you receive.

You can pass any number of fields inside the schema (that might not actually exist in the form) but the source of truth of _real_ fields rendered in the form is the DOM (JSX) --> so when a field mounts and say to the form it is there. This is why I don't rely on the schema to return defaultValue because the field might not exist.

For now, the consumer has to do the check manually:

```js
if (title === undefined) {
  return;
}
```

I'll give it more thought to improve this unexpected behaviour 👍

Reply:
If resolving the behavior is more difficult than expected it would be good to document it.